### ISSUE TITLE
Set settings in `my.cfg`

### ISSUE THREAD

**Issue by abitrolly** [ansible-collections/ansible.mysql#284]
##### SUMMARY

It looks like [these modules](https://docs.ansible.com/ansible/latest/collections/community/mysql/index.html) provide no way to set global `mysqld` settings in `/etc/my.cfg`, like `skip-name-resolve` etc.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
community.mysql

##### ADDITIONAL INFORMATION

There should be a least an example how to use other modules and locate necessary `my.cfg` files to add such settings.

**Comment by Andersson007** [comment]
@abitrolly thanks for sharing the idea!

Do I understand correctly that you need to add some settings to `/etc/my.cnf`?
Can't we achieve that by using some core modules like [lineinfile](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/lineinfile_module.html), for example?

**Comment by abitrolly** [comment]
@Andersson007 yes, I needed to set the config variable as a workaround for this bug - https://mariadb.zulipchat.com/#narrow/stream/118759-general/topic/Insufficient.20data.20for.20.22Access.20denied.22.20errors

The problem is that I am not certain which config file is being used, and where is the section that should be updated.

**Comment by laurent-indermuehle** [comment]
@abitrolly this is typically something that you would address with an Ansible role. Roles are not yet widely available in collections. Mine is in a private repo for instance. I think it's because roles are often highly opinionated and hard to share.

MySQL and MariaDB can include sub-config files that overwrites variables already present on the main config. In your main config [/etc/my.cnf] you'll find a line `!includedir /etc/my.cnf.d`. This makes it easy to add your configuration in a file generated using the module `ansible.builtin.template` that you write inside that [/etc/my.cnf.d] folder.

In conclusion, I don't think writing a module is necessary for mysql/mariadb. They both supports the "drop-in directories" ending in ".d".
